engine/src/main/java/org/pentaho/di/job/entries/deleteresultfilenames/JobEntryDeleteResultFilenames.java

```diff
@@ -0,0 +1,258 @@
+    super( n, "" );
+    foldername = null;
+    wildcardexclude = null;
+    wildcard = null;
+    specifywildcard = false;
+  }
+
+  public JobEntryDeleteResultFilenames() {
+    this( "" );
+  }
+
+  public Object clone() {
+    JobEntryDeleteResultFilenames je = (JobEntryDeleteResultFilenames) super.clone();
+    return je;
+  }
+
+  public String getXML() {
+    StringBuilder retval = new StringBuilder( 100 ); // 75 chars in just tag names and spaces
+
+    retval.append( super.getXML() );
+    retval.append( "      " ).append( XMLHandler.addTagValue( "foldername", foldername ) );
+    retval.append( "      " ).append( XMLHandler.addTagValue( "specify_wildcard", specifywildcard ) );
+    retval.append( "      " ).append( XMLHandler.addTagValue( "wildcard", wildcard ) );
+    retval.append( "      " ).append( XMLHandler.addTagValue( "wildcardexclude", wildcardexclude ) );
+
+    return retval.toString();
+  }
+
+  public void loadXML( Node entrynode, List<DatabaseMeta> databases, List<SlaveServer> slaveServers,
+    Repository rep, IMetaStore metaStore ) throws KettleXMLException {
+    try {
+      super.loadXML( entrynode, databases, slaveServers );
+      foldername = XMLHandler.getTagValue( entrynode, "foldername" );
+      specifywildcard = "Y".equalsIgnoreCase( XMLHandler.getTagValue( entrynode, "specify_wildcard" ) );
+      wildcard = XMLHandler.getTagValue( entrynode, "wildcard" );
+      wildcardexclude = XMLHandler.getTagValue( entrynode, "wildcardexclude" );
+
+    } catch ( KettleXMLException xe ) {
+      throw new KettleXMLException( BaseMessages.getString(
+        PKG, "JobEntryDeleteResultFilenames.CanNotLoadFromXML", xe.getMessage() ) );
+    }
+  }
+
+  public void loadRep( Repository rep, IMetaStore metaStore, ObjectId id_jobentry, List<DatabaseMeta> databases,
+    List<SlaveServer> slaveServers ) throws KettleException {
+    try {
+      foldername = rep.getJobEntryAttributeString( id_jobentry, "foldername" );
+      specifywildcard = rep.getJobEntryAttributeBoolean( id_jobentry, "specify_wildcard" );
+      wildcard = rep.getJobEntryAttributeString( id_jobentry, "wildcard" );
+      wildcardexclude = rep.getJobEntryAttributeString( id_jobentry, "wildcardexclude" );
+    } catch ( KettleException dbe ) {
+      throw new KettleXMLException( BaseMessages.getString(
+        PKG, "JobEntryDeleteResultFilenames.CanNotLoadFromRep", "" + id_jobentry, dbe.getMessage() ) );
+    }
+  }
+
+  public void saveRep( Repository rep, IMetaStore metaStore, ObjectId id_job ) throws KettleException {
+    try {
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "foldername", foldername );
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "specify_wildcard", specifywildcard );
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "wildcard", wildcard );
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "wildcardexclude", wildcardexclude );
+    } catch ( KettleDatabaseException dbe ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)